picasso/src/main/java/com/squareup/picasso/ExifStreamReader.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2010 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+ * Adapted for picasso 
+ */
+
+package com.squareup.picasso;
+
+import android.util.Log;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class ExifStreamReader {
+  private static final String TAG = "CameraExif";
+
+  // Returns the orientation value
+  public static int getOrientation(InputStream stream) throws IOException {
+    if (stream == null) {
+      return 0 ;
+    }
+    MarkableInputStream markStream = new MarkableInputStream(stream);
+    long mark = markStream.savePosition(65536); 
+    byte[] header = new byte[65536];
```

Comment:
We can't do this kind of allocation. I have a better solution in mind but not the time to implement it. I have a flight on Sunday. I'll try to do it then.